postmortem/2023-10-10.md

```diff
@@ -0,0 +1,107 @@
+---
+Postmortem ID: 6
+Title: Stake duration bonus not enabled for nft bucket
+Author: Chen Chen <chenchen@iotex.me>
+Created: 2023-10-10
+Updated: 2023-11-09
+---
+
+
+# Abstract
+
+During the testing of the Staking as NFT(IIP-13) feature on Mainnet, it was found that the vote calculation of the NFT Bucket was incorrect. This resulted in a slight decrease in the calculation of delegate total votes and the distribution of bucket rewards. To fix this issue, a hard fork upgrade is required.
+
+# Cause
+
+The votes for the NFT Bucket only calculate the staked amount and does not consider the weighting of stake duration and stake-lock.
+
+**Incorrect calculation**
+
+$$votes=amount$$
+
+**The correct**
+
+$$votes=amount*(1+\log_{1.2}{(duration/86400)(1+m)})$$
+
+$$m= \begin{cases} 1 & locked \\ 0 & unlocked \geq 1 \end{cases}$$
+
+# Solution
+
+Using the accurate formula for vote calculation post hard-fork height. State patching is unnecessary as NFT votes are calculated in real-time, instead of being stored in the database. Also ensure the consistency of the vote calculation formula across blockchain, rewarding service, and web/mobile.
+
+We calculate the rewards for NFT staking from epoch 37021 to 39613 based on the new reward rules and provide users with a one-time supplemental payment for the difference. The details are shown below:
```

Comment:
explain 37021 is when IIP-13 contract deployed, and 39613 is when hard-fork activated